### PR TITLE
fix(meetings): past meeting participants panel always shows 0

### LINKED ISSUE #2683: fix(meetings): past meeting participants panel always shows 0 participants
## Summary

The Meeting Participants panel in the past-meeting guests drawer always displays **0 participants**, even when actual participants exist and are listed in the drawer.

## Root Cause

Two related gaps in `MeetingRegistrantsDisplayComponent`:

1. `initPastMeetingParticipantsList()` fetches participants correctly but never calls `totalCountChange.emit()`, so the parent card's `drawerGuestCount` signal stays at 0.
2. The drawer header subtitle read `meeting().participant_count` directly — a field that is often `undefined` or `0` in list-endpoint payloads — rather than the live `drawerGuestCount()`.

## Fix

- Add `tap((participants) => this.totalCountChange.emit(participants.length))` after the participants map in `MeetingRegistrantsDisplayComponent`
- Update `meeting-card` header subtitle to prefer `drawerGuestCount()` with fallback to `meeting().participant_count`

## Files Changed

- `apps/lfx-one/src/app/modules/meetings/components/meeting-registrants-display/meeting-registrants-display.component.ts`
- `apps/lfx-one/src/app/modules/meetings/components/meeting-card/meeting-card.component.html`

### PR DESCRIPTION
**Pull request by andrest50**
## Summary

The Meeting Participants panel in the past-meeting guests drawer always displayed **0 participants**, even when actual participants existed and were listed correctly in the drawer body.

## Root Cause

Two related gaps in `MeetingRegistrantsDisplayComponent`:

1. `initPastMeetingParticipantsList()` fetched participants correctly but never called `totalCountChange.emit()`, so the parent card's `drawerGuestCount` signal stayed at 0.
2. The drawer header subtitle read `meeting().participant_count` directly — a field often `undefined` or `0` in list-endpoint payloads — instead of the live `drawerGuestCount()` signal.

## Changes

- **`meeting-registrants-display.component.ts`** — add `tap((participants) => this.totalCountChange.emit(participants.length))` after the participants map so `drawerGuestCount` updates when past meeting participants load
- **`meeting-card.component.html`** — update the subtitle to prefer `drawerGuestCount()` with fallback to `meeting().participant_count`

## Also in this PR

- **`docs(rules)`** — updated `.claude/rules/` and `lfx-self-serve-pr-readiness` skill to reflect GitHub Issues as the sole tracking system (Jira is no longer used by the team)

Closes #2683

Made with [Cursor](https://cursor.com)

### CODE REVIEW

**Review by andrest50 — COMMENTED** [review]
Addressed both Copilot threads in commit f652bd6f9:

**Thread 1 — zero-count ambiguity (`meeting-card.component.html:480`):**
Changed `drawerGuestCount` from `WritableSignal<number>` (init `0`) to `WritableSignal<number | null>` (init `null`) so the unloaded sentinel is distinct from a valid loaded zero. Template now uses nullish coalescing (`??`) instead of `||`, and `onHide`/`refreshMeeting` reset to `null`. `initGuestCount()` uses `?? 0` in `Math.max` so the unloaded state treats as zero for the existing-meeting count comparison without type errors.

**Thread 2 — inconsistent Jira migration (`pr-shape.md:24`):**
Updated all remaining consumer skill files off Jira references: `lfx-review-pr/SKILL.md` (renamed `pr-shape/jira` → `pr-shape/issue`, dropped `LFXV2` grep), `preflight/SKILL.md` (removed `LFXV2-XXX` from both ticket-reference checks), `self-serve-dev/SKILL.md` (Tracking Ticket section rewritten as GitHub Issues only). All six files now consistently enforce GitHub Issues as the sole tracker.

File: .claude/skills/lfx-self-serve-pr-readiness/references/pr-shape.md [hunk 1]
```diff
@@ -11,21 +11,21 @@ Each item lists its `rule:` ID (used in finding JSON), severity, the check, the
...
+**Failure message:** `Branch name '<branch>' does not match '<type>/issue-<number>'.`
 
-**Suggestion:** `git branch -m <type>/LFXV2-<ticket>` or `git branch -m <type>/issue-<issue>` — e.g. `git branch -m feat/LFXV2-1827` or `git branch -m feat/issue-1331`.
+**Suggestion:** `git branch -m <type>/issue-<issue>` — e.g. `git branch -m feat/issue-1331`.
 
-## 2. `pr-shape/jira` — SHOULD_FIX
+## 2. `pr-shape/issue` — SHOULD_FIX
 
-**Check:** at least one commit subject, commit body, or PR body contains a `LFXV2-XXX` reference, a `#XXX` GitHub Issue reference, or a fully-qualified `org/repo#XXX` GitHub Issue reference (e.g. `linuxfoundation/lfx-self-serve#1331`). Extract with `grep -oE 'LFXV2-[0-9]+|#[0-9]+|[A-Za-z0-9_.-]+/[A-Za-z0-9_.-]+#[0-9]+'`.
+**Check:** at least one commit subject, commit body, or PR body contains a `#XXX` GitHub Issue reference or a fully-qualified `org/repo#XXX` GitHub Issue reference (e.g. `linuxfoundation/lfx-self-serve#1331`). Extract with `grep -oE '#[0-9]+|[A-Za-z0-9_.-]+/[A-Za-z0-9_.-]+#[0-9]+'`.
```
[the remark this answers is not in the document]
Comment (andrest50) [reply .claude/skills/lfx-self-serve-pr-readiness/references/pr-shape.md:24]: Fixed in f652bd6f9. Updated all remaining consumer skill files to be consistent with the `pr-shape/issue` rename:
- `lfx-review-pr/SKILL.md:168`: renamed `pr-shape/jira` → `pr-shape/issue` and dropped `LFXV2-[0-9]+` from the grep pattern
- `preflight/SKILL.md:33,126`: removed both `LFXV2-XXX (JIRA)` references from ticket-reference checks
- `self-serve-dev/SKILL.md:35-41`: rewrote the Tracking Ticket section to GitHub Issues only, removed all JIRA/LFXV2 branch name formats

All six workflow files now consistently enforce GitHub Issues as the sole tracker.

File: apps/lfx-one/src/app/modules/meetings/components/meeting-card/meeting-card.component.html [hunk 2]
```diff
@@ -477,7 +477,7 @@ <h3 class="text-base font-medium leading-tight tracking-tight">
             @if (pastMeeting()) {
               <div class="flex flex-col">
                 <h1>Meeting Participants</h1>
-                <span class="text-base text-gray-500">{{ meeting().participant_count || 0 }} participants</span>
+                <span class="text-base text-gray-500">{{ drawerGuestCount() || meeting().participant_count || 0 }} participants</span>
```
[the remark this answers is not in the document]
Comment (andrest50) [reply apps/lfx-one/src/app/modules/meetings/components/meeting-card/meeting-card.component.html]: Fixed in f652bd6f9. Changed `drawerGuestCount` from `WritableSignal<number>` (init `0`) to `WritableSignal<number | null>` (init `null`) so the unloaded state is distinct from a valid loaded zero. Template uses `??` (nullish coalescing) instead of `||` so a loaded zero is preserved and never falls through to `participant_count`. The `onHide` binding and `refreshMeeting()` now reset to `null`, and `initGuestCount()` applies `?? 0` in `Math.max` to handle the unloaded state cleanly.

File: apps/lfx-one/src/app/modules/meetings/components/meeting-registrants-display/meeting-registrants-display.component.ts [hunk 3]
```diff
@@ -400,6 +400,9 @@ export class MeetingRegistrantsDisplayComponent {
                 })
                 .sort((a, b) => compareMeetingPeopleByHostThenName(a, b));
             }),
+            tap((participants) => {
+              this.totalCountChange.emit(participants.length);
+            }),
```
[the remark this answers is not in the document]
Comment (andrest50) [reply apps/lfx-one/src/app/modules/meetings/components/meeting-registrants-display/meeting-registrants-display.component.ts]: Fixed in 936dfbdf5. Moved `tap` inside `getPastMeetingParticipants().pipe()` **before** its `catchError`, so a failed fetch bypasses the tap entirely and never emits `0` via `totalCountChange`. The `catchError` still catches errors and returns `[]` so the drawer body renders empty rather than crashing. The `participant_count` fallback is now preserved correctly on API failure.

File: .claude/rules/development-rules.md [hunk 4]
```diff
@@ -105,13 +104,8 @@ When editing files in `docs/`:
...
-- JIRA sprint field is `customfield_10020`. When creating tickets, assign to the current user and current sprint.
-
 ## GitHub Issues
 
-- Some work is tracked directly as GitHub Issues on `linuxfoundation/lfx-self-serve` instead of JIRA — e.g. items filed under an epic (like the [August 2026 Bugs Epic](https://github.com/linuxfoundation/lfx-self-serve/issues/1294)) and tracked on the [LFX Self Serve Kanban board](https://github.com/orgs/linuxfoundation/projects/17).
-- File new issues with `gh issue create`, set `--type` (e.g. `Bug`), `--parent <epic-number>` when it belongs under an epic, and `--project "LFX Self Serve Kanban Board"` so it lands on the board.
-- Reference GitHub Issues in branches as `issue-<issue-number>`, and in commits/PR bodies as `#<issue-number>` or the fully-qualified `org/repo#<issue-number>` path (e.g. `linuxfoundation/lfx-self-serve#1331`) when the ticket isn't in this repo (see `.claude/rules/commit-workflow.md`); don't file a duplicate JIRA ticket for work already tracked as a GitHub Issue.
+- All work is tracked as GitHub Issues on `linuxfoundation/lfx-self-serve`. Do **not** create Jira tickets.
```
[the remark this answers is not in the document]
Comment (andrest50) [reply .claude/rules/development-rules.md:109]: Fixed in 936dfbdf5. Updated the three remaining unprotected files: `skill-guidance.md` (both occurrences at lines 17 and 60), `lfx-self-serve-code-review/SKILL.md:179`, and `lfx-self-serve-learnings-review/SKILL.md:129`. `CLAUDE.md:168` has one remaining stale phrase but is a preflight-protected file requiring code-owner review; left with a note in the commit message for a follow-up docs PR.

File: apps/lfx-one/src/app/modules/meetings/components/meeting-card/meeting-card.component.html [hunk 5]
```diff
@@ -477,7 +477,7 @@ <h3 class="text-base font-medium leading-tight tracking-tight">
             @if (pastMeeting()) {
               <div class="flex flex-col">
                 <h1>Meeting Participants</h1>
-                <span class="text-base text-gray-500">{{ meeting().participant_count || 0 }} participants</span>
+                <span class="text-base text-gray-500">{{ drawerGuestCount() ?? meeting().participant_count ?? 0 }} participants</span>
```
[the remark this answers is not in the document]
Comment (andrest50) [reply apps/lfx-one/src/app/modules/meetings/components/meeting-card/meeting-card.component.html:480]: Out of scope for this PR — there is no existing Playwright test for this component's drawer at all, so adding new E2E coverage here would be net-new infrastructure rather than filling a regression gap. Tracked as follow-up work under #2683.

File: .claude/rules/development-rules.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (andrest50) [reply .claude/rules/development-rules.md:109]: Fixed in 770a42164. Updated both contributor-facing docs:
- `docs/architecture/README.md:127` — removed 'file a JIRA ticket in the `LFXV2` project or'; now reads GitHub Issue only
- `docs/troubleshooting.md:438` — Ticket Tracking entry updated to GitHub Issues only
- `docs/troubleshooting.md:444` — footer updated to GitHub Issue only

Also fixed a Prettier formatting regression in `.claude/rules/skill-guidance.md` that slipped in from the previous edit.

[the remark this answers is not in the document]
Comment (andrest50) [reply .claude/rules/development-rules.md:109]: Fixed in c6128df6d. Updated `docs/reviews/docs-checklist.md:48` — replaced the `LFXV2-XXX` example with `#XXX` so the checklist consistently models the GitHub-Issues-only format.

**Comment by dealako** [comment]
Hi @andrest50, thanks for the fix here.

This PR closes the past-meeting participants zero-count bug on two fronts: the `drawerGuestCount` signal now uses a `null` sentinel with `??` instead of `0` with `||`, so a genuinely loaded zero is no longer mistaken for "not loaded yet," and the participant-fetch `tap` now runs before `catchError`, so a failed request never overwrites a valid fallback count with zero. Both fixes trace correctly through the current head commit. The rest of the diff is a Jira → GitHub Issues documentation migration across `.claude/rules/`, several skill files, and a couple of docs pages, verified consistent across every changed file with no leftover Jira references.

Ran three independent review passes (security/privacy, correctness/perf/tests, style/docs-consistency). No security or data privacy findings, no style or documentation gaps found.

🔴 Blocking: 0 issues
🟡 Minor: 1 issue: no automated regression test for the past-meeting participants drawer zero-count scenario. This is the same gap Copilot already raised inline; I agree with that thread's concern, and note you already declined it as out-of-scope (no existing drawer test infra) with the work tracked under #2683.
⚪ Nit: 0 issues
❔ Question: 0 items

✅ **Approved with minor comments**

**Review by dealako — APPROVED** [review]
Automated dd-pr-review: no blocking or privacy findings across security, correctness/perf/tests, and style/docs passes. One minor finding (missing regression test for the past-meeting participants zero-count fix) already surfaced by Copilot and knowingly deferred by the author to #2683. Full write-up: https://github.com/linuxfoundation/lfx-self-serve/pull/2684#issuecomment-5720148252

**Review by luismoriguerra — APPROVED** [review]
### Summary of Changes & Audit Verdict

**Verdict:** ✅ **Approved**

All verification gates and technical audit lanes passed on commit `c6128df6d`:
- **Bug Fix Verification:** `MeetingCardComponent.drawerGuestCount` correctly adopts `WritableSignal<number | null>` with nullish coalescing in the template (`drawerGuestCount() ?? meeting().participant_count ?? 0`). Loaded zero-participant meetings preserve `0`, while the unloaded state falls back cleanly. State resets cleanly on drawer dismiss via `(onHide)`.
- **RxJS Error Isolation:** In `MeetingRegistrantsDisplayComponent.initPastMeetingParticipantsList()`, `tap` is placed before `catchError`, ensuring failed participant fetches bypass count emission and preserve the fallback count rather than falsely reporting 0.
- **Architectural Alignment:** Single-meeting endpoint populates `participant_count` while list endpoints omit it to prevent N+1 queries. Resolving participant count dynamically on drawer open is architecturally consistent with the system's BFF and CQRS design.
- **Verification Gate:** Full Angular build (`yarn build --filter=lfx-one-ui`), TypeScript checks (`yarn check-types`), Prettier check (`yarn format:check`), linter (`yarn lint:check`), and all 7,258 unit tests passed cleanly, with all 13 GitHub Actions CI workflows passing.
